discovery-provider/src/tasks/index_aggregate_track.py

```diff
@@ -0,0 +1,167 @@
+import logging
+
+from src.tasks.celery_app import aggregate_worker, celery, celery_worker
+from src.tasks.index_aggregate_user import get_latest_blocknumber
+
+logger = logging.getLogger(__name__)
+
+AGGREGATE_TRACK = "aggregate_track"
+
+# UPDATE_AGGREGATE_TRACK_QUERY
+# Get a higher bound blocknumber to check for new save/repost activity
+# Find a subset of tracks that have changed after that blocknumber
+# For that subset of tracks calculate the play/repost total counts
+# Insert that count for new users or update an existing row
+UPDATE_AGGREGATE_TRACK_QUERY = """
+    WITH aggregate_track_checkpoints AS (
+        SELECT
+            :current_blocknumber AS current_blocknumber
+    ),
+    new_track AS (
+        SELECT
+            t.track_id AS track_id
+        FROM
+            tracks t
+        GROUP BY
+            t.track_id
+        UNION
+        ALL (
+            SELECT
+                r.repost_item_id AS track_id
+            FROM
+                reposts r
+            WHERE
+                r.is_current IS TRUE
+                AND r.repost_type = 'track'
+                AND r.is_delete IS FALSE
+                AND r.blocknumber <= (
+                    SELECT
+                        current_blocknumber
+                    FROM
+                        aggregate_track_checkpoints
+                )
+            GROUP BY
+                r.repost_item_id
+        )
+        UNION
+        ALL (
+            SELECT
+                s.save_item_id AS track_id
+            FROM
+                saves s
+            WHERE
+                s.is_current IS TRUE
+                AND s.save_type = 'track'
+                AND s.is_delete IS FALSE
+                AND s.blocknumber <= (
+                    SELECT
+                        current_blocknumber
+                    FROM
+                        aggregate_track_checkpoints
+                )
+            GROUP BY
+                s.save_item_id
+        )
+    )
+    INSERT INTO
+        aggregate_track (track_id, repost_count, save_count)
+    SELECT
+        DISTINCT(t.track_id),
+        COALESCE(track_repost.repost_count, 0) AS repost_count,
+        COALESCE(track_save.save_count, 0) AS save_count
+    FROM
+        (
+            SELECT
+                t.track_id
+            FROM
+                tracks t
+            GROUP BY
+                t.track_id
+        ) AS t
+        LEFT OUTER JOIN (
+            SELECT
+                r.repost_item_id AS track_id,
+                count(r.repost_item_id) AS repost_count
+            FROM
+                reposts r
+            WHERE
+                r.is_current IS TRUE
+                AND r.repost_type = 'track'
+                AND r.is_delete IS FALSE
+                AND r.repost_item_id IN (
+                    SELECT
+                        track_id
+                    FROM
+                        new_track
+                )
+                AND r.blocknumber <= (
```

Comment:
Similar to aggregate_user we don't actually want to filter here. Idea is for any tracks that have changed, we'll want to recount all of its reposts and saves. This avoids any accidental misses or double counts.